File: codesignal/challenges/mostFrequentDigitSum/mostFrequentDigitSum.py
def mostFrequentDigitSum(n):
    r = [sum_digit(n)]
    while n > 0:
        x = n - sum_digit(n)
        r.append(sum_digit(x))
        n = x

    r = sorted(r)
    return most_frequent(r[::-1])

def sum_digit(n):
    s = 0
    for i in str(n):
        s += int(i)
    return s


def most_frequent(List):
    counter = 0
    num = List[0]

    for i in List:
        curr_frequency = List.count(i)
        if(curr_frequency> counter):
            counter = curr_frequency
            num = i

    return num

File: codesignal/challenges/mostFrequentDigitSum/mostFrequentDigitSum.rs
fn sum_digit(n: i32) -> i32 {
    let s = n.to_string();
    let mut sum = 0;
    for i in 0..s.len() {
        let x = &s[i..i+1];
        sum += x.parse::<i32>().unwrap();
    }
    return sum;
}

fn most_frequent(v: Vec<i32>) -> i32 {
    let mut result = v[0];
    let mut count = 0;
    for num in v.iter() {
        let current_frequency = v.iter().filter(|&n| *n == *num).count();
        if current_frequency > count {
            result = *num;
            count = current_frequency;
        }
    }
    return result;
}

fn mostFrequentDigitSum(n: i32) -> i32 {
    let mut s = vec![sum_digit(n)];
    let mut m = n;
    while m > 0 {
        let temp = m - sum_digit(m);
        s.push(sum_digit(temp));
        m = temp;
    }
    s.sort();
    s.reverse();
    return most_frequent(s);
}

